controllers/loadtest_controller_test.go

```diff
@@ -1,850 +1,309 @@
-			build.Image = &customImage
-
-			container := newBuildContainer(build)
-			Expect(container.Image).To(Equal(customImage))
-		})
-
-		It("sets command", func() {
-			command := []string{"bazel"}
-			build.Command = command
-
-			container := newBuildContainer(build)
-			Expect(container.Command).To(Equal(command))
-		})
-
-		It("sets args", func() {
-			args := []string{"build", "//target"}
-			build.Command = []string{"bazel"}
-			build.Args = args
-
-			container := newBuildContainer(build)
-			Expect(container.Args).To(Equal(args))
-		})
-
-		It("sets environment variables", func() {
-			env := []corev1.EnvVar{
-				{Name: "EXPERIMENT", Value: "1"},
-				{Name: "PROD", Value: "0"},
-			}
-
-			build.Env = env
 
-			container := newBuildContainer(build)
-			Expect(env[0]).To(BeElementOf(container.Env))
-			Expect(env[1]).To(BeElementOf(container.Env))
-		})
-	})
-
-	Describe("newRunContainer", func() {
-		var run grpcv1.Run
-
-		BeforeEach(func() {
-			image := "docker.pkg.github.com/grpc/test-infra/fake-image"
-			command := []string{"qps_worker"}
-
-			run = grpcv1.Run{
-				Image:   &image,
-				Command: command,
-			}
-		})
-
-		It("sets the name of the container", func() {
-			container := newRunContainer(run)
-			Expect(container.Name).To(Equal(config.RunContainerName))
-		})
-
-		It("sets bazel cache volume mount", func() {
-			container := newRunContainer(run)
-			volumeMount := newBazelCacheVolumeMount()
-			Expect(container.VolumeMounts).To(ContainElement(volumeMount))
-		})
-
-		It("sets workspace volume mount", func() {
-			container := newRunContainer(run)
-			volumeMount := newWorkspaceVolumeMount()
-			Expect(container.VolumeMounts).To(ContainElement(volumeMount))
-		})
-
-		It("sets workspace as working directory", func() {
-			container := newRunContainer(run)
-			Expect(container.WorkingDir).To(Equal(config.WorkspaceMountPath))
-		})
-
-		It("sets image", func() {
-			image := "golang:1.14"
-			run.Image = &image
-
-			container := newRunContainer(run)
-			Expect(container.Image).To(Equal(image))
-		})
-
-		It("sets command", func() {
-			command := []string{"go"}
-			run.Command = command
-
-			container := newRunContainer(run)
-			Expect(container.Command).To(Equal(command))
-		})
-
-		It("sets args", func() {
-			command := []string{"go"}
-			args := []string{"run", "main.go"}
-			run.Command = command
-			run.Args = args
-
-			container := newRunContainer(run)
-			Expect(container.Args).To(Equal(args))
-		})
-
-		It("sets environment variables", func() {
-			env := []corev1.EnvVar{
-				{Name: "ENABLE_DEBUG", Value: "1"},
-				{Name: "VERBOSE", Value: "1"},
+			for i := range list.Items {
+				item := &list.Items[i]
+				if item.Labels[config.LoadTestLabel] == test.Name {
+					foundPodCount++
+				}
 			}
 
-			run.Env = env
-
-			container := newRunContainer(run)
-			Expect(env[0]).To(BeElementOf(container.Env))
-			Expect(env[1]).To(BeElementOf(container.Env))
-		})
+			return foundPodCount, nil
```

Comment:
If the function returns 2 values and the expectedPodCount is just 1 value, how would they match? Will the nil not be counted?